pkg/pillar/cmd/tpmmgr/tpmmgr.go

```diff
@@ -719,12 +725,16 @@ func DecryptSecretWithEcdhKey(X, Y *big.Int, iv, ciphertext, plaintext []byte) e
 }
 
 // getDecryptKey : uses the ECC params to construct the AES decryption Key
-func getDecryptKey(X, Y *big.Int) ([32]byte, error) {
-	// when TPM is not enabled, use the locally stored private key
-	if !etpm.IsTpmEnabled() {
-		privateKey, err := getDevicePrivateKey()
+func getDecryptKey(X, Y *big.Int, eveNodeCert *types.EveNodeCert) ([32]byte, error) {
+	// check whether this is a software certificate
+	if !etpm.IsTpmEnabled() || !eveNodeCert.IsTpm {
+		isDeviceKey := false
+		if !etpm.IsTpmEnabled() && eveNodeCert == nil {
+			isDeviceKey = true
+		}
```

Comment:
Is this to handle old devices? Wouldn't they get a ECDH certificate created on boot when they run this new code?
In any case, if this code needs to stay it needs comments explaining why one would ever use the deviceKey for ECDH.

Reply:
Added the comments.